src/module.api/cache/defid.cache.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { Test } from '@nestjs/testing'
+import { CacheModule } from '@nestjs/common'
+import { DeFiDCache, CachePrefix } from '@src/module.api/cache/defid.cache'
+import { createToken, createPoolPair } from '@defichain/testing'
+import { PoolPairInfo } from '@defichain/jellyfish-api-core/dist/category/poolpair'
+import { TokenInfo } from '@defichain/jellyfish-api-core/dist/category/token'
+
+const container = new MasterNodeRegTestContainer()
+let client: JsonRpcClient
+let cache: DeFiDCache
+
+describe('getPoolPairInfo', () => {
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    client = new JsonRpcClient(await container.getCachedRpcUrl())
+
+    const module = await Test.createTestingModule({
+      imports: [CacheModule.register()],
+      providers: [
+        { provide: JsonRpcClient, useValue: client },
+        DeFiDCache
+      ]
+    }).compile()
+
+    cache = module.get(DeFiDCache)
+
+    for (const token of ['A', 'B', 'C']) {
+      await container.waitForWalletBalanceGTE(110)
+      await createToken(container, token)
+    }
+
+    await createPoolPair(container, 'A', 'B') // 4
+    await createPoolPair(container, 'A', 'C') // 5
+    await createPoolPair(container, 'B', 'C') // 6
+
+    // TODO(canonbrother): add listpoolpairs in @defi/testing
+    const poolPairResult = await container.call('listpoolpairs')
+    for (const k in poolPairResult) {
+      await cache.getPoolPairInfo(k)
+    }
+
+    await container.stop()
+  })
+
+  it('should get from cache via get as container RPC is killed', async () => {
+    const abKey = `${CachePrefix.POOL_PAIR_INFO} 4`
+    const abPoolPair = await cache.testGet<PoolPairInfo>(abKey)
```

Comment:
```suggestion
    const cache = module.get<Cache>(Cache)
    const abPoolPair = await cache.get<PoolPairInfo>(abKey)
```

Should be able to return to do this.